Controller/Midpage/Cancel.php

```diff
@@ -0,0 +1,43 @@
+<?php
+namespace Billink\Billink\Controller\Midpage;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Payment\Gateway\Data\PaymentDataObjectInterface;
+
+class Cancel extends AbstractAction
+{
+    /**
+     * @return
+     */
+    public function execute()
+    {
+        $resultRedirect = $this->redirectFactory->create();
+
+        $transactionOrder = (string)$this->getOrderFromTransaction();
```

Comment:
Typecast is unnecessary?

Reply:
No, it was before this function was implemented :)